### PR TITLE
Added Consul and Zk2 topo support in Go Cluster endtoend tests

### PR DESCRIPTION
**Pull request by ajeetj**
Moved etcd_process.go to topo_process.go
Added support for zk2 and consul
Added Start/Shutdown methods for zk2 and consul
Handle --topo-flavor flag [default to etcd2]

### CODE REVIEW

**Comment by ajeetj** [comment]
Looks like the test failed in github actions. I will fix those tomorrow.

File: go/test/endtoend/cluster/topo_process.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+	if flavor == "consul" {
+		binary = "consul"
+	}
+
+	topo := &TopoProcess{
+		Name:     name,
+		Binary:   binary,
+		Port:     port,
+		Host:     hostname,
+		PeerPort: peerPort,
```
Comment (deepthi) [inline go/test/endtoend/cluster/topo_process.go]: do we need to store both PeerPort and PeerURL?

Comment (ajeetj) [reply go/test/endtoend/cluster/topo_process.go]: @deepthi  Yes, etcd required these ports. 
Python also uses two ports but they name them as clientPort and peerPort.

Comment (ajeetj) [reply go/test/endtoend/cluster/topo_process.go]: Right, we only need PeerURL in this case. I will remove the extra variable.

**Pull request merged**